core/cautils/strutils_test.go

```diff
@@ -135,3 +135,59 @@ func TestStringSlicesAreEqual(t *testing.T) {
 		})
 	}
 }
+
+func TestParseBoolEnvVar(t *testing.T) {
+	testCases := []struct {
+		expectedErr  string
+		name         string
+		varName      string
+		varValue     string
+		defaultValue bool
+		expected     bool
+	}{
+		{
+			name:         "Variable does not exist",
+			varName:      "DOES_NOT_EXIST",
+			varValue:     "",
+			defaultValue: true,
+			expected:     true,
+			expectedErr:  "",
+		},
+		{
+			name:         "Variable exists and is a valid bool",
+			varName:      "MY_VAR",
+			varValue:     "true",
+			defaultValue: false,
+			expected:     true,
+			expectedErr:  "",
+		},
+		{
+			name:         "Variable exists but is not a valid bool",
+			varName:      "MY_VAR",
+			varValue:     "not_a_boolean",
+			defaultValue: false,
+			expected:     false,
+			expectedErr:  "failed to parse MY_VAR env var as bool",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.varValue != "" {
+				os.Setenv(tc.varName, tc.varValue)
+			} else {
+				os.Unsetenv(tc.varName)
+			}
```

Comment:
nitpick: `os.Setenv` actually sets the environment variable on the system

We should use `testing.T.Setenv` instead. It will set and unset the environment variable in the scope of the test run.

Reply:
Thanks for the advice, I'll update